indexing-service/src/main/java/org/apache/druid/indexing/input/DruidSegmentReader.java

```diff
@@ -70,9 +68,8 @@
 import java.util.NoSuchElementException;
 import java.util.Set;
 
-public class DruidSegmentReader extends IntermediateRowParsingReader<Map<String, Object>>
+public class DruidSegmentReader extends DruidSegmentReaderBase
```

Comment:
Design nit: I'm not sure that this parent class is really adding much.  You haven't been able to delete much code from this class, it looks like all you are doing is delegating the storage of `source`.  For the TombstoneReader, it's also not really doing that much.  Seems like it might even be less code to have them just be separate classes without any attempt at inheritance.

Reply:
Removed base class